src/components/UnstyledLink/UnstyledLink.tsx

```diff
@@ -29,3 +25,8 @@ export const UnstyledLink = React.memo(
     );
   }),
 );
+
+// Wrapping forwardRef in a memo means the component does not get a name set
+// So we have to do it ourselves
+// https://github.com/facebook/react/issues/16722
```

Comment:
Bug whack-a-mole. I'm not convinced we should be wrapping this in memo in the first place: https://github.com/Shopify/polaris-react/pull/2330